magefiles/rulesengine/repos/infra_deployments.go

```diff
@@ -1,17 +1,194 @@
+	Description: "Run the Konflux-demo suite tests when an Infra-deployments PR includes changes to files outside of the specified components.",
+	Condition: rulesengine.None{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule,
+		rulesengine.ConditionFunc(CheckNoFilesChanged)},
+
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(ExecuteInfraDeploymentsDefaultTestAction)}}
+
+// ExecuteInfraDeploymentsDefaultTestAction excutes all the e2e-tests and component suites
+func ExecuteInfraDeploymentsDefaultTestAction(rctx *rulesengine.RuleCtx) error {
+	rctx.LabelFilter = "konflux"
+	return ExecuteTestAction(rctx)
 }
+
+// InfraDeploymentsTestFilesOnlyRule defines rules of test suites running of each changed component
+var InfraDeploymentsTestFilesOnlyRule = rulesengine.Rule{Name: "Infra-deployments PR Components File Diff Execution",
+	Description: "Runs specific tests of changed component by infra-deployments PR.",
+	Condition: rulesengine.Any{
+		&InfraDeploymentsIntegrationTestFileChangeRule,
+		&InfraDeploymentsImageControllerTestFileChangeRule,
+		&InfraDeploymentsMultiPlatformTestFileChangeRule,
+		&InfraDeploymentsBuildTemplatesTestFileChangeRule,
+		&InfraDeploymentsBuildServiceTestFileChangeRule,
+		&InfraDeploymentsReleaseServiceTestFileChangeRule,
+		&InfraDeploymentsEnterpriseControllerTestFileChangeRule,
+		&InfraDeploymentsJVMTestFileChangeRule},
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		// Adding konflux-demo to the label filtetr when componenets changed
+		if rctx.LabelFilter == "" {
+			rctx.LabelFilter = "konflux"
+		} else {
+			rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "konflux")
+		}
+		return nil
+
+	}),
+		rulesengine.ActionFunc(ExecuteTestAction)}}
+
+var InfraDeploymentsIntegrationTestFileChangeRule = rulesengine.Rule{Name: "Infra-deployments PR Integration component File Change Rule",
+	Description: "Map Integration tests files when Integration component files are changed in the infra-deployments PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("components/integration/*")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+		if !strings.Contains(rctx.LabelFilter, "integration-service") {
+			if rctx.LabelFilter == "" {
+				rctx.LabelFilter = "integration-service"
+			} else {
+				rctx.LabelFilter = fmt.Sprintf("%s,%s", rctx.LabelFilter, "integration-service")
+			}
+		}
```

Comment:
this would be worth to put into reusable function

Reply:
created function to implement this solution